UPGRADING.md

```diff
@@ -1,6 +1,300 @@
+For more information about all the updates, please refer to the [release notes](https://fast-dds.docs.eprosima.com/en/latest/notes/notes.html).
 
-When upgrading from a version older than 1.7.0, regeneration is **required**.
+As announced in previous minor releases, it is always *recommended* to regenerate the types with the latest *eProsima Fast DDS Gen* release.
+In this case, the type regeneration is required, using *Fast DDS Gen* v4.0.0.
+
+The following sections describe the possible changes that your project may require to migrate to *Fast DDS v3.0.0*.
+
+- [Library management](#library-management)
+- [Compatibility with Fast CDR](#compatibility-with-fast-cdr)
+- [Namespace migrations and changes](#namespace-migrations-and-changes)
+- [Public headers migrated to *fastdds*](#public-headers-migrated-to-fastdds)
+- [Public headers moved to private](#public-headers-moved-to-private)
+- [API changes](#api-changes)
+- [Struct, Enum, Variable](#struct-enum-variable)
+- [Examples](#examples)
+
+
+## Library management
+
+The below list expose the changes related to the package name, environment variables and other library usages.
+
+* The package has been renamed from `fastrtps` to `fastdds`.
+* XML profiles loading environment variable has been renamed to: `FASTDDS_DEFAULT_PROFILES_FILE`.
+* The configuration file that Fast DDS looks for to load the profiles has been renamed to `DEFAULT_FASTDDS_PROFILES.xml`.
+* XML Schema namespace in fastdds_profiles.xsd has been updated to http://www.eprosima.com.
+* CMake Windows file names have been changed:
+
+    * fastdds.manifest.in
+    * fastdds-config.cmake
+
+* The fastrtps.rc file has been renamed as fastdds.rc.
+* Deprecated APIs marked with `FASTDDS_DEPRECATED` and `FASTDDS_TODO_BEFORE` macros have been removed.
+
+## Compatibility with Fast CDR
+
+Fast DDS v3 is only compatible with Fast CDR v2.
+If you are not using Fast CDR as [third-party](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html#third-party-libraries-options), please ensure updating your local dependencies.
+
+## Namespace migrations and changes
+
+The following list contains the namespace changes and migrations:
+
+* All `eprosima::fastrtps::` namespace references were moved to `eprosima::fastdds::`.
+* `SubscriptionBuiltinTopicData`, `PublicationBuiltinTopicData` and `ParticipantBuiltinTopicData` were moved from `fastdds::dds::builtin::` to `fastdds::dds::`.
+* `EventKindBits::` references changed to`EventKind::`.
+* `EventKindEntityId::` references changed to`EntityId::`.
+* `StatisticsEventKind::` references changed to `statistics::EventKind::`.
+* `Duration_t` and `c_TimeInfinite` reference were moved to `dds::`.
+* `ParticipantAttributes`, `PublisherAttributes`, `SubscriberAttributes`, `ReplierAttributes`, `RequesterAttributes` were moved to `xmlparser::`.
+* `Time_t.hpp` references were explicitly distingueshed between `dds::Time_t::` and `rtps::Time_t::`.
```

Comment:
We already had two classes, I think what we need to say is that they are in different files now